server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -823,6 +823,14 @@ public String getWriteableName() {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
+            if (out.getTransportVersion().before(TransportVersions.REGEX_AND_RANGE_INTERVAL_QUERIES)) {
+                throw new UnsupportedOperationException(
```

Comment:
this seems a little redundant with the gating of this feature behind a capability - assuming that the capability ensures that all nodes in the cluster have it ?

Reply:
As  I understood, a cluster can have nodes with different capabilities. But I don't understand is how does this gating even work? I think it is still possible for a node to send this query request to another node  that doesn't have this capability?  

Also, what is the benefit of throwing this error now, as opposed on a shard which will happen automatically where a shard discovers an unknown NamedWriteableRegistry.Entry?

Reply:
This check should be handled automatically by implementing `VersionedNamedWriteable`, and implementing the `getMinimalSupportedVersion` method